Products/ZenUtils/zenpack.py

```diff
@@ -481,6 +481,10 @@ def doRestore(pack):
         fixedSomething = len(sortedPacks) > 0
         while len(sortedPacks) > 0:
             packListLen = len(sortedPacks)
+            # Load zenpacks files before normal installation to avoid
+            # upgrade breaking (ZEN-24274).
+            for packs in sortedPacks:
+                self._loadFiles(packs, zpsToRestore[packs][0])
```

Comment:
This `for` loop and the outer `while` loop are both iterating over `sortedPacks`.  I think this will result in many repetitive files-only installs of ZenPacks.  Perhaps the files-only install of `sortedPacks` should occur before the `while` loop.